src/DynamoCore/Models/CustomNodeWorkspaceModel.cs

```diff
@@ -196,20 +196,23 @@ protected virtual void OnFunctionIdChanged(Guid oldId)
 
         public override bool SaveAs(string newPath, ProtoCore.RuntimeCore runtimeCore, bool isBackUp = false)
         {
-            var originalPath = FileName;
-
-            // A SaveAs to an existing function id prompts the creation of a new 
-            // custom node with a new function id
-            if (originalPath != newPath)
+            if (!isBackUp)
```

Comment:
The code changes can be less (and easier to reason about) if we do early return:

```csharp
if (isBackup)
    return base.SaveAs(newPath, runtimeCore);
```

Does that make sense?